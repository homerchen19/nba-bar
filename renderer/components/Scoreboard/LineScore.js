import React from 'react';
import PropTypes from 'prop-types';
import { Flex } from 'antd-mobile';
import styled from 'styled-components';
import { StickyTable, Row, Cell } from 'react-sticky-table';

import { colors } from '@styles/theme';
import nba from '@utils/nba';

const Wrapper = styled(Flex)`
  width: 100%;
  background: #fff;
`;

const StyledCell = styled(Cell)`
  padding: 8px 12px;
  border-bottom: 1px solid ${colors.white};
  text-align: center;
`;

const TeamName = styled.p`
  color: #fff;
  font-weight: 500;
`;

const reanderTeamRow = team => (
  <Row key={team.name}>
    <StyledCell
      key="name"
      style={{
        backgroundColor: nba.getTeamBackgroundColor(team.name),
        padding: '8px 16px',
      }}
    >
      <TeamName>{team.name}</TeamName>
    </StyledCell>
    {team.linescores.map(period => (
      <StyledCell key={period.period_value}>{period.score}</StyledCell>
    ))}
    <StyledCell key="score">{team.score}</StyledCell>
  </Row>
);

const LineScore = ({ home, visitor }) => (
  <Wrapper justify="start">
    <StickyTable stickyHeaderCount={0}>
      <Row key="header">
        <StyledCell key="name" />
        {home.linescores.map(period => (
          <StyledCell key={period.period_value}>
            <b>{period.period_name}</b>
          </StyledCell>
        ))}
        <StyledCell key="total">
          <b>T</b>
        </StyledCell>
      </Row>
      {reanderTeamRow(home, true)}
      {reanderTeamRow(visitor, false)}
    </StickyTable>
  </Wrapper>
);

LineScore.propTypes = {
  home: PropTypes.object.isRequired,
  visitor: PropTypes.object.isRequired,
};

export default LineScore;
