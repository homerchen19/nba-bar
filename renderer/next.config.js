const withCSS = require('@zeit/next-css');
const withImages = require('next-images');

module.exports = withImages(
  withCSS({
    cssModules: true,
    webpack(config, { isServer }) {
      config.target = 'electron-renderer';

      if (isServer) {
        const antStyles = /antd-mobile\/.*?\/style.*?/;
        const origExternals = [...config.externals];
        config.externals = [
          (context, request, callback) => {
            if (request.match(antStyles)) return callback();
            if (typeof origExternals[0] === 'function') {
              origExternals[0](context, request, callback);
            } else {
              callback();
            }
          },
          ...(typeof origExternals[0] === 'function' ? [] : origExternals),
        ];

        config.module.rules.unshift({
          test: antStyles,
          use: 'null-loader',
        });
      }

      return config;
    },
    exportPathMap() {
      return {
        '/home': { page: '/home' },
        '/live': { page: '/live' },
        '/preview': { page: '/preview' },
        '/scoreboard': { page: '/scoreboard' },
        '/settings': { page: '/settings' },
        '/standings': { page: '/standings' },
      };
    },
  })
);
