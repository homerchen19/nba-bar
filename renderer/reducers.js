import { combineReducers } from 'redux';

import homeReducer from './containers/Home/reducer';
import previewReducer from './containers/Preview/reducer';
import liveReducer from './containers/Live/reducer';
import scoreboardReducer from './containers/Scoreboard/reducer';
import standingsReducer from './containers/Standings/reducer';
import settingsReducer from './containers/Settings/reducer';

const rootReducer = combineReducers({
  home: homeReducer,
  preview: previewReducer,
  live: liveReducer,
  scoreboard: scoreboardReducer,
  standings: standingsReducer,
  settings: settingsReducer,
});

export default rootReducer;
