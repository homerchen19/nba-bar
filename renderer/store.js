import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';

import rootReducer from './reducers';

const middleware = [thunk, logger];

const enhancers = compose(
  applyMiddleware(...middleware),
  typeof window !== 'undefined' &&
    window.devToolsExtension &&
    process.env.NODE_ENV !== 'production'
    ? window.devToolsExtension()
    : f => f
);

export default function configureStore(initialState = {}) {
  const store = createStore(rootReducer, initialState, enhancers);

  return store;
}
