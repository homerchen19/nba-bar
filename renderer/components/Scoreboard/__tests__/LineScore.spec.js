import React from 'react';
import renderer from 'react-test-renderer';

import LineScore from '../LineScore';

const realMath = Object.create(global.Math);

beforeAll(() => {
  const mockMath = Object.create(global.Math);
  mockMath.random = () => 0.5;
  global.Math = mockMath;
});

afterAll(() => {
  global.Math = realMath;
});

it('be defined', () => {
  expect(LineScore).toBeDefined();
});

it('match snapshot', () => {
  const props = {
    home: {
      name: 'IND',
      score: '126',
      linescores: [
        {
          period_name: 'Q1',
          period_value: '1',
          score: '27',
        },
        {
          period_name: 'Q2',
          period_value: '2',
          score: '35',
        },
        {
          period_name: 'Q3',
          period_value: '3',
          score: '35',
        },
        {
          period_name: 'Q4',
          period_value: '4',
          score: '29',
        },
      ],
    },
    visitor: {
      name: 'GSW',
      score: '106',
      linescores: [
        {
          period_name: 'Q1',
          period_value: '1',
          score: '26',
        },
        {
          period_name: 'Q2',
          period_value: '2',
          score: '25',
        },
        {
          period_name: 'Q3',
          period_value: '3',
          score: '26',
        },
        {
          period_name: 'Q4',
          period_value: '4',
          score: '29',
        },
      ],
    },
  };
  const tree = renderer.create(<LineScore {...props} />).toJSON();

  expect(tree).toMatchSnapshot();
});
