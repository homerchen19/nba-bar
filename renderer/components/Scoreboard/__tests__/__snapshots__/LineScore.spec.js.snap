// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  className="am-flexbox LineScore__Wrapper-sc-16p4iy3-0 iMqDAJ am-flexbox-justify-start am-flexbox-align-center"
>
  <div
    className="sticky-table "
  >
    <div
      className="x-scrollbar"
    >
      <div />
    </div>
    <div
      className="y-scrollbar"
    >
      <div />
    </div>
    <div
      className="sticky-table-header-wrapper"
    >
      <div
        className="sticky-table-corner hidden"
      >
        <div
          className="sticky-table-table "
        />
      </div>
      <div
        className="sticky-table-header hidden"
      >
        <div
          className="sticky-table-table "
        />
      </div>
    </div>
    <div
      className="sticky-table-y-wrapper"
    >
      <div
        className="sticky-table-column "
      >
        <div
          className="sticky-table-table "
        >
          <div
            className="sticky-table-row sticky-table-row "
            id=""
          >
            <div
              className="sticky-table-cell LineScore__StyledCell-sc-16p4iy3-1 cZnJd"
            />
          </div>
          <div
            className="sticky-table-row sticky-table-row "
            id=""
          >
            <div
              className="sticky-table-cell LineScore__StyledCell-sc-16p4iy3-1 cZnJd"
              style={
                Object {
                  "backgroundColor": "#041e42",
                  "padding": "8px 16px",
                }
              }
            >
              <p
                className="LineScore__TeamName-sc-16p4iy3-2 jLhbNS"
              >
                IND
              </p>
            </div>
          </div>
          <div
            className="sticky-table-row sticky-table-row "
            id=""
          >
            <div
              className="sticky-table-cell LineScore__StyledCell-sc-16p4iy3-1 cZnJd"
              style={
                Object {
                  "backgroundColor": "#ffc72d",
                  "padding": "8px 16px",
                }
              }
            >
              <p
                className="LineScore__TeamName-sc-16p4iy3-2 jLhbNS"
              >
                GSW
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        className="sticky-table-x-wrapper"
      >
        <div
          className="sticky-table-table "
        >
          <div
            className="sticky-table-row sticky-table-row "
            id=""
          >
            <div
              className="sticky-table-cell LineScore__StyledCell-sc-16p4iy3-1 cZnJd"
            >
              <b>
                Q1
              </b>
            </div>
            <div
              className="sticky-table-cell LineScore__StyledCell-sc-16p4iy3-1 cZnJd"
            >
              <b>
                Q2
              </b>
            </div>
            <div
              className="sticky-table-cell LineScore__StyledCell-sc-16p4iy3-1 cZnJd"
            >
              <b>
                Q3
              </b>
            </div>
            <div
              className="sticky-table-cell LineScore__StyledCell-sc-16p4iy3-1 cZnJd"
            >
              <b>
                Q4
              </b>
            </div>
            <div
              className="sticky-table-cell LineScore__StyledCell-sc-16p4iy3-1 cZnJd"
            >
              <b>
                T
              </b>
            </div>
          </div>
          <div
            className="sticky-table-row sticky-table-row "
            id=""
          >
            <div
              className="sticky-table-cell LineScore__StyledCell-sc-16p4iy3-1 cZnJd"
            >
              27
            </div>
            <div
              className="sticky-table-cell LineScore__StyledCell-sc-16p4iy3-1 cZnJd"
            >
              35
            </div>
            <div
              className="sticky-table-cell LineScore__StyledCell-sc-16p4iy3-1 cZnJd"
            >
              35
            </div>
            <div
              className="sticky-table-cell LineScore__StyledCell-sc-16p4iy3-1 cZnJd"
            >
              29
            </div>
            <div
              className="sticky-table-cell LineScore__StyledCell-sc-16p4iy3-1 cZnJd"
            >
              126
            </div>
          </div>
          <div
            className="sticky-table-row sticky-table-row "
            id=""
          >
            <div
              className="sticky-table-cell LineScore__StyledCell-sc-16p4iy3-1 cZnJd"
            >
              26
            </div>
            <div
              className="sticky-table-cell LineScore__StyledCell-sc-16p4iy3-1 cZnJd"
            >
              25
            </div>
            <div
              className="sticky-table-cell LineScore__StyledCell-sc-16p4iy3-1 cZnJd"
            >
              26
            </div>
            <div
              className="sticky-table-cell LineScore__StyledCell-sc-16p4iy3-1 cZnJd"
            >
              29
            </div>
            <div
              className="sticky-table-cell LineScore__StyledCell-sc-16p4iy3-1 cZnJd"
            >
              106
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
